src/build.ts

```diff
@@ -149,7 +149,8 @@ function outputPlugin(output: Output, ext: string, base: Plugin) {
       return resolveId ? resolveId.call(this, source, base, options) : undefined
     },
     load(file: string) {
-      return output.get(file) || (load && load.call(this, file))
+      let code = output.get(file);
+      return code ? {code, map: output.get(file + '.map')} : (load && load.call(this, file));
```

Comment:
Instead of assuming the source map for a file is the same as the name of file with a .map extension, a more general approach would be to scan the code for sourceMapping comment to extract the sourcemap file, for eg using ([ source-map-resolve](https://github.com/lydell/source-map-resolve)). I went with the current approach as it is simple and performant and seems to work perfectly fine for this case